RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -209,15 +220,149 @@ class EvtPlaneProducer : public edm::stream::EDProducer<> {
+        if (track.eta < EPEtaMin1[i])
+          continue;
+        if (track.eta > EPEtaMax2[i])
+          continue;
+        if (track.eta > EPEtaMax1[i] && track.eta < EPEtaMin2[i])
+          continue;
+      }
+      double w = track.et;
+      if (loadDB_)
+        w = track.et * flat[i]->getEtScale(vz, bin);
+      if (EPOrder[i] == 1) {
+        if (MomConsWeight[i][0] == 'y' && loadDB_) {
+          w = flat[i]->getW(track.et, vz, bin);
+        }
+        //if(track.eta<0 ) w=-w;
+      }
+      rp[i]->addParticle(w, track.et, sin(EPOrder[i] * track.phi), cos(EPOrder[i] * track.phi), track.eta);
+    }
+  };
+
+  void FillCastor(TrackStructure track, double vz, int bin) {
+    double minet = minet_;
+    double maxet = maxet_;
+    for (int i = 0; i < NumEPNames; i++) {
+      if (EPDet[i] == Castor) {
+        if (minet_ < 0)
+          minet = minTransverse[i];
+        if (maxet_ < 0)
+          maxet = maxTransverse[i];
+        if (track.et < minet)
+          continue;
+        if (track.et > maxet)
+          continue;
+        if (EPEtaMin2[i] == EPEtaMax2[i]) {
+          if (track.eta < EPEtaMin1[i])
+            continue;
+          if (track.eta > EPEtaMax1[i])
+            continue;
+        } else {
+          if (track.eta < EPEtaMin1[i])
+            continue;
+          if (track.eta > EPEtaMax2[i])
+            continue;
+          if (track.eta > EPEtaMax1[i] && track.eta < EPEtaMin2[i])
+            continue;
+        }
+        double w = track.et;
+        if (EPOrder[i] == 1) {
+          if (MomConsWeight[i][0] == 'y' && loadDB_) {
+            w = flat[i]->getW(track.et, vz, bin);
+          }
+          //if(track.eta<0 ) w=-w;
+        }
+        rp[i]->addParticle(w, track.et, sin(EPOrder[i] * track.phi), cos(EPOrder[i] * track.phi), track.eta);
+      }
+    }
+  }
+
+  void FillTracker(TrackStructure track, double vz, int bin) {
+    double minpt = minpt_;
+    double maxpt = maxpt_;
+    for (int i = 0; i < NumEPNames; i++) {
+      if (EPDet[i] == Tracker) {
+        if (minpt_ < 0)
+          minpt = minTransverse[i];
+        if (maxpt_ < 0)
+          maxpt = maxTransverse[i];
+        if (track.pt < minpt)
+          continue;
+        if (track.pt > maxpt)
+          continue;
+        if (EPEtaMin2[i] == EPEtaMax2[i]) {
+          if (track.eta < EPEtaMin1[i])
+            continue;
+          if (track.eta > EPEtaMax1[i])
+            continue;
+        } else {
+          if (track.eta < EPEtaMin1[i])
+            continue;
+          if (track.eta > EPEtaMax2[i])
+            continue;
+          if (track.eta > EPEtaMax1[i] && track.eta < EPEtaMin2[i])
+            continue;
+        }
+        double w = track.pt;
+        if (w > 2.5)
+          w = 2.0;  //v2 starts decreasing above ~2.5 GeV/c
+        if (EPOrder[i] == 1) {
+          if (MomConsWeight[i][0] == 'y' && loadDB_) {
+            w = flat[i]->getW(track.pt, vz, bin);
+          }
+          //if(track.eta<0) w=-w;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant